### PR TITLE
added mirroring functions

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

File: src/sempy_labs/_mirrored_databases.py [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+    for r in responses:
+        for v in r.get("value", []):
+            prop = v.get("properties", {})
+            sql = prop.get("sqlEndpointProperties", {})
+            new_data = {
+                "Mirrored Database Name": v.get("displayName"),
+                "Mirrored Database Id": v.get("id"),
+                "Description": v.get("description"),
+                "OneLake Tables Path": (
+                    prop.get("oneLakeTablesPath") if prop is not None else None
```
Comment (eisber) [inline src/sempy_labs/_mirrored_databases.py:64]: I don't think prop can be None

File: src/sempy_labs/_mirrored_databases.py [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+    )
+
+    for r in responses:
+        for v in r.get("data", []):
+            m = v.get("metrics", {})
+            new_data = {
+                "Source Schema Name": v.get("sourceSchemaName"),
+                "Source Table Name": v.get("sourceTableName"),
+                "Status": v.get("status"),
+                "Processed Bytes": m.get("processedBytes") if m is not None else None,
```
Comment (eisber) [inline src/sempy_labs/_mirrored_databases.py:254]: m can't be None?

**Review by eisber — CHANGES_REQUESTED** [review]

File: src/sempy_labs/_mirrored_databases.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/_mirrored_databases.py]: fixed

File: src/sempy_labs/_mirrored_databases.py (hunk 2 above)
Comment (m-kovalsky) [reply src/sempy_labs/_mirrored_databases.py]: fixed

**Review by eisber — APPROVED** [review]

**Pull request merged**